src/Lachain.Core/Blockchain/VM/ExternalHandler.cs

```diff
@@ -378,6 +378,53 @@ public static void Handler_Env_GetMsgValue(int dataOffset)
             if (!ret)
                 throw new InvalidContractException("Bad call to (get_msg_value)");
         }
+        
+        public static void Handler_Env_GetExternalBalance(int addressOffset, int resultOffset)
+        {
+            Logger.LogInformation($"Handler_Env_GetExternalBalance({addressOffset}, {resultOffset})");
+            var frame = VirtualMachine.ExecutionFrames.Peek() as WasmExecutionFrame
+                        ?? throw new InvalidOperationException("Cannot call GetExternalBalance outside wasm frame");
+            
+            // Get the address from the given memory offset
+            var snapshot = frame.InvocationContext.Snapshot;
+            var addressBuffer = SafeCopyFromMemory(frame.Memory, addressOffset, 20);
+            if (addressBuffer is null)
+                throw new InvalidContractException("Bad call to call function");
```

Comment:
I think it is better to have more specific message here:
```suggestion
                throw new InvalidContractException("Bad call to (get_external_balance)");
```

Reply:
incorporated the comments